### PR TITLE
adds pull to refresh feature in tableView

### PR DESCRIPTION
**Pull request by cordhernandez**
BlackSourceLabs/BlackNectar-iOS#23

### CODE REVIEW

File: BlackNectar/BlackNectar/StoresTableViewController.swift [hunk 1]
```diff
@@ -34,13 +36,14 @@ class StoresTableViewController: UITableViewController, SideMenuFilterDelegate {
         
     }()
     
-    private let main = OperationQueue.main
+    fileprivate let main = OperationQueue.main
```
Comment (SirWellington) [inline BlackNectar/BlackNectar/StoresTableViewController.swift:38]: nice!

File: BlackNectar/BlackNectar/StoresTableViewController.swift [hunk 2]
```diff
@@ -190,3 +192,44 @@ class StoresTableViewController: UITableViewController, SideMenuFilterDelegate {
     }
     
 }
+
+extension StoresTableViewController {
+    
+    func setupRefreshControl() {
```
Comment (SirWellington) [inline BlackNectar/BlackNectar/StoresTableViewController.swift:197]: I love how you put this in separate extension! A+

File: BlackNectar/BlackNectar/StoresTableViewController.swift [hunk 3]
```diff
@@ -190,3 +192,44 @@ class StoresTableViewController: UITableViewController, SideMenuFilterDelegate {
     }
     
 }
+
+extension StoresTableViewController {
```
Comment (SirWellington) [inline BlackNectar/BlackNectar/StoresTableViewController.swift:195]: By the way, if nothing in this extension will be used outside this file, then you can mark the extension as `fileprivate`. Then all of the methods defined inside of it are not visible outside this file.

File: BlackNectar/BlackNectar/StoresTableViewController.swift [hunk 4]
```diff
@@ -190,3 +192,44 @@ class StoresTableViewController: UITableViewController, SideMenuFilterDelegate {
...
+    
+    func setupRefreshControl() {
+        
+        refreshControl = UIRefreshControl()
+        refreshLoadingView = UIView(frame: self.refreshControl!.bounds)
```
Comment (SirWellington) [inline BlackNectar/BlackNectar/StoresTableViewController.swift]: This is a case where forced unwrap is acceptable, though I would still do a `guard let` statement anyway.

File: BlackNectar/BlackNectar/StoresTableViewController.swift [hunk 5]
```diff
@@ -190,3 +192,44 @@ class StoresTableViewController: UITableViewController, SideMenuFilterDelegate {
...
+        
+        refreshLoadingView.backgroundColor = UIColor.clear
```
Comment (SirWellington) [inline BlackNectar/BlackNectar/StoresTableViewController.swift]: Don't really see you using this `refreshLoadingView` anywhere. You could probably do without it.

File: BlackNectar/BlackNectar/StoresTableViewController.swift [hunk 6]
```diff
@@ -190,3 +192,44 @@ class StoresTableViewController: UITableViewController, SideMenuFilterDelegate {
...
+    }
+    
+    
+    func reloadStoreData() {
+        
+        guard let usersLocation = UserLocation.instance.currentCoordinate else { return }
+        let usersLatitude = usersLocation.latitude
+        let usersLongitude = usersLocation.longitude
+        
+        SearchStores.searchForStoresLocations(near: usersLocation) { stores in
```
Comment (SirWellington) [inline BlackNectar/BlackNectar/StoresTableViewController.swift:216]: This will probably conflict with some of Kevin's changes.

**Review by SirWellington — APPROVED** [review]
Nice and simple. You are on-point with these small pull requests. Love it. 

You should merge your PR in before Kevin does.

File: BlackNectar/BlackNectar/StoresTableViewController.swift (hunk 3 above)
Comment (cordhernandez) [reply BlackNectar/BlackNectar/StoresTableViewController.swift:195]: I'm thinking that we might need to have some form of refresh feature for the mapView? Another topic we need to discuss with Courtney?

File: BlackNectar/BlackNectar/StoresTableViewController.swift (hunk 2 above)
Comment (cordhernandez) [reply BlackNectar/BlackNectar/StoresTableViewController.swift:197]: ooohhhh Tank yuuuu

File: BlackNectar/BlackNectar/StoresTableViewController.swift (hunk 5 above)
Comment (cordhernandez) [reply BlackNectar/BlackNectar/StoresTableViewController.swift]: copy that 
successfully executed

File: BlackNectar/BlackNectar/StoresTableViewController.swift (hunk 4 above)
Comment (cordhernandez) [reply BlackNectar/BlackNectar/StoresTableViewController.swift]: on the job
refreshLoadingView is now removed

File: BlackNectar/BlackNectar/StoresTableViewController.swift (hunk 6 above)
Comment (cordhernandez) [reply BlackNectar/BlackNectar/StoresTableViewController.swift:216]: Ok conflict will be resolved 
will discuss this with Kevin

**Pull request merged**